src/snowflake/connector/ocsp_snowflake.py

```diff
@@ -59,39 +56,57 @@
 )
 from snowflake.connector.errors import RevocationCheckError
 from snowflake.connector.network import PYTHON_CONNECTOR_USER_AGENT
-from snowflake.connector.ssd_internal_keys import (
-    ocsp_internal_dep1_key_ver,
-    ocsp_internal_dep2_key_ver,
-    ocsp_internal_ssd_pub_dep1,
-    ocsp_internal_ssd_pub_dep2,
-)
 from snowflake.connector.telemetry_oob import TelemetryService
 from snowflake.connector.time_util import DecorrelateJitterBackoff
 
 from . import constants
 from .cache import SFDictCache, SFDictFileCache
 from .telemetry import TelemetryField, generate_telemetry_data_dict
 
+
+class OCSPResponseValidationMetadata(NamedTuple):
+    exception: Exception | None = None
+    issuer: Certificate | None = None
+    subject: Certificate | None = None
+    cert_id: CertId | None = None
+    ocsp_response: bytes | None = None
+
+
+class OCSPResponseValidationResult(NamedTuple):
+    ocsp_response_validation_metadata: OCSPResponseValidationMetadata | None = None
+    ts: int | None = None
+    validated: bool = False
+
+
 try:
-    OCSP_CACHE: SFDictFileCache[
-        tuple[bytes, bytes, int, str],
-        tuple[Exception | None, Certificate, Certificate, CertId, bytes],
+    OCSP_RESPONSE_VALIDATION_CACHE: SFDictFileCache[
+        tuple[bytes, bytes, int],
+        OCSPResponseValidationResult,
     ] = SFDictFileCache(
         entry_lifetime=constants.DAY_IN_SECONDS,
         file_path={
-            "linux": os.path.join("~", ".cache", "snowflake", "ocsp_cache"),
-            "darwin": os.path.join("~", "Library", "Caches", "Snowflake", "ocsp_cache"),
+            "linux": os.path.join(
+                "~", ".cache", "snowflake", "ocsp_response_validation_cache"
+            ),
+            "darwin": os.path.join(
+                "~", "Library", "Caches", "Snowflake", "ocsp_response_validation_cache"
+            ),
             "windows": os.path.join(
-                "~", "AppData", "Local", "Snowflake", "Caches", "ocsp_cache"
+                "~",
+                "AppData",
+                "Local",
+                "Snowflake",
+                "Caches",
+                "ocsp_response_validation_cache",
             ),
         },
     )
 except OSError:
     # In case we run into some read/write permission error fall back onto
     #  in memory caching
-    OCSP_CACHE: SFDictCache[
-        tuple[bytes, bytes, int, str],
-        tuple[Exception | None, Certificate, Certificate, CertId, bytes],
+    OCSP_RESPONSE_VALIDATION_CACHE: SFDictCache[
+        tuple[bytes, bytes, int],
```

Comment:
You modified [generate_cache_key's signature](https://github.com/snowflakedb/snowflake-connector-python/blob/3a94c4a1c07f460e07d09f051e0cd4725a7b51e0/src/snowflake/connector/ocsp_snowflake.py#L119), based on that I believe this should be
```suggestion
        tuple[bytes, bytes, bytes]
```

Reply:
fixed